arelle/ValidateXbrl.py

```diff
@@ -680,6 +683,21 @@ def checkLinks(self, modelLinks: set[ModelLink]) -> None:
                                 linkrole=modelLink.role,
                                 xlinkLabel=resourceArcToLabel)
 
+    def checkDuplicateFacts(self, facts: list[ModelFact], duplicateType: ValidateDuplicateFacts.DuplicateType) -> None:
+        duplicateFactSets = ValidateDuplicateFacts.getDuplicateFactSetsOfType(facts, duplicateType)
+        if not duplicateFactSets:
+            return
+        for duplicateFactSet in duplicateFactSets:
+            self.modelXbrl.error(
+                "arelle:disallowedDuplicateFacts",
+                "Duplicate facts (%(duplicateType)s) %(qname)s: %(values)s, Example pair: %(contextIDs)s.",
+                modelObject=duplicateFactSet.facts,
```

Comment:
Does the duplicate fact set potentially include inconsistent duplicates for complete (and consistent) validation if there are both complete and inconsistent duplicates?

For example:
* duplicate fact check mode is set to complete.
* there are 3 duplicate facts (factA, factB, and factC.)
* factA and factB are complete duplicates.
* factC is an inconsistent duplicate of factA and factB.
* is factC logged here?

Reply:
Or does the inconsistent factC cause the two complete duplicates not to be logged?

Reply:
As discussed separately, this PR has been updated to treat the incoming argument as an "any" check, i.e. if any complete, consistent, incomplete, inconsistent facts exist within a set, the set should be logged.